src/Services/PantheonGuzzle.php

```diff
@@ -153,14 +153,21 @@ public function getPsr18Adapter(): AdapterInterface {
    * @return \Psr\Http\Message\RequestInterface
    */
   public function requestUriAlterForPantheonEnvironment(RequestInterface $r) {
+    $toAdd = '';
     $uri = $r->getUri();
     $path = $uri->getPath();
-    $endpointHasPath = strpos($path, $this->endpoint->getPath());
-    if (strpos($path, $this->endpoint->getPath()) === FALSE) {
-      $uri = $uri->withPath($this->endpoint->getPath() . $this->endpoint->getCore() . $path);
-      return $r->withUri($uri);
+    $exploded = explode('/', $path);
+    if (!in_array(trim($this->endpoint->getCore(), '/'), $exploded)) {
```

Comment:
Is `getCore()` guaranteed to never contain a `/`? Could you change the code to continue to use strpos on getCore and getPath, or was there some reason you switched to looking at the elements of the paths individually with in_array?

Reply:
it trims first and last. The main thing is the solarium library inconsistently has some path/handler values that start and/or end with "/" and some that do not. Trying to establish that all path/handler values that are sent to another function do not have a start and ending "/" so that they can be added anywhere without getting a double "//"

Reply:
Yes, but the old code works with paths like `foo/bar`, whereas the new code would fail. If `getCore()` never contains an *interior* `/`, then that's fine; in that instance, the new code is clearer. The old code is safer if there might ever in the future be an interior `/` in the `getCore()` path.

Reply:
Changed it to search for site_id in the exploded array.